File: frontend/src/api/speech.ts
import type { SpeechSynthesisRequest, SpeechSynthesisResponse } from '../types/chat'

const API_BASE_URL = import.meta.env.VITE_API_BASE_URL ?? 'http://127.0.0.1:8000'

export async function synthesizeSpeech(request: SpeechSynthesisRequest) {
  const response = await fetch(`${API_BASE_URL}/api/speech/synthesize`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(request)
  })

  if (!response.ok) {
    throw new Error(`语音合成请求失败，状态码：${response.status}`)
  }

  return (await response.json()) as SpeechSynthesisResponse
}

File: backend/app/routers/speech.py
from fastapi import APIRouter, HTTPException, status

from app.config import settings
from app.schemas import SpeechSynthesisRequest, SpeechSynthesisResponse
from app.services.aliyun_tts_service import (
    AliyunTTSConfigurationError,
    AliyunTTSService,
    AliyunTTSUpstreamError,
)


router = APIRouter(prefix="/api/speech", tags=["speech"])
tts_service = AliyunTTSService()


@router.post("/synthesize", response_model=SpeechSynthesisResponse)
async def synthesize_speech(request: SpeechSynthesisRequest) -> SpeechSynthesisResponse:
    try:
        audio_url, audio_format = await tts_service.synthesize(request.text)
    except AliyunTTSConfigurationError as error:
        raise HTTPException(
            status_code=status.HTTP_503_SERVICE_UNAVAILABLE,
            detail="后端未配置阿里云 API Key，请检查 ALIYUN_API_KEY。",
        ) from error
    except AliyunTTSUpstreamError as error:
        raise HTTPException(
            status_code=status.HTTP_502_BAD_GATEWAY,
            detail="阿里云语音合成调用失败，已回退到浏览器播报。",
        ) from error

    return SpeechSynthesisResponse(
        audio_url=audio_url,
        format=audio_format,
        model=settings.aliyun_tts_model,
    )
